src/main/java/life/qbic/domain/project/ProjectRepository.java

```diff
@@ -0,0 +1,39 @@
+package life.qbic.domain.project;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.inject.Singleton;
+import life.qbic.domain.sample.SampleCode;
+import life.qbic.domain.sample.SampleEvent;
+import life.qbic.domain.sample.SampleEventDatasource;
+
+/**
+ * ProjectRepository used to read project information from persistence.
+ *
+ * @since 2.0.0
+ */
+@Singleton
+public class ProjectRepository {
+
+  private final SampleEventDatasource sampleEventDatasource;
+
+  public ProjectRepository(SampleEventDatasource sampleEventDatasource) {
+    this.sampleEventDatasource = sampleEventDatasource;
+  }
+
+  public Optional<Project> get(ProjectCode projectCode) {
+    List<SampleEvent> allMatchingProject = sampleEventDatasource.findAllMatchingProject(
```

Comment:
NIT: I was at first confused what "allMatchingProject"means. 
Since we're retrieving all sample events associated with a project code this could be more distinct
```suggestion
    List<SampleEvent> SamplesMatchingProject = sampleEventDatasource.findSamplesMatchingProject(
```
However i like that you adhered to the event naming [convention](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-events)

Reply:
As this is an SampleEventDatasource, if it is read completely it is quite direct.
`sample event datasource find all matching project`

Reply:
Thanks for explaining! 👍 Since this is a NIT this should not block the PR anyway :)